tests/third_party/cupy/math_tests/test_matmul.py

```diff
@@ -128,6 +129,15 @@ def test_overlap_both(self, xp, dtype, shape):
         return xp.matmul(a, a, out=a)
 
```

Comment:
In `TestMatmulOut` there are shapes skipped: `((0,), (0,), (0,)),`, but if to enable that it will trigger an issue, because:
```python
a = numpy.ones((0,))
out = numpy.zeros((0,))

numpy.matmul(a, a, out=out)
# Out: array([], dtype=float64)

ia = dpnp.array(a)
iout = dpnp.array(out)

dpnp.matmul(ia, ia, out=iout)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[7], line 1
----> 1 dpnp.matmul(ia, ia, out=iout)

File ~/code/dpnp/dpnp/dpnp_iface_linearalgebra.py:421, in matmul(x1, x2, out, casting, order, dtype, subok, signature, extobj, axes, axis)
    418             # out that is passed to the backend should have the correct shape
    419             out = dpnp.moveaxis(out, axes_res, (-2, -1))
--> 421 result = dpnp_matmul(
    422     x1,
    423     x2,
    424     out=out,
    425     casting=casting,
    426     order=order,
    427     dtype=dtype,
    428 )
    429 if axes is not None:
    430     if out is result:
    431         # out and out_orig contain the same data but they have different shape

File ~/code/dpnp/dpnp/dpnp_utils/dpnp_utils_linearalgebra.py:501, in dpnp_matmul(x1, x2, out, casting, order, dtype)
    499         return result
    500 else:
--> 501     return dpnp.get_result_array(result, out, casting=casting)

File ~/code/dpnp/dpnp/dpnp_iface.py:503, in get_result_array(a, out, casting)
    501 dpnp.check_supported_arrays_type(out)
    502 if out.shape != a.shape:
--> 503     raise ValueError(
    504         f"Output array of shape {a.shape} is needed, got {out.shape}."
    505     )
    506 elif isinstance(out, dpt.usm_ndarray):
    507     out = dpnp_array._create_from_usm_ndarray(out)

ValueError: Output array of shape () is needed, got (0,).
```
I guess we need to resolve that or to address.